### PR TITLE
FEATURE: NAS-179 Add data sampling to charts

### PR DESCRIPTION
**Pull request by Tulean**
JIRA: NAS-179

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by Tulean** [comment]
extended check sonar

**Comment by Tulean** [comment]
extended test

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1551

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4492/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/321/ : SUCCESS in 11m 24s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4491/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6112/ : SUCCESS in 13m 54s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1081/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4493/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1081/ : SUCCESS in 18m 27s

File: libs/sdk-ui-ext/src/insightView/InsightView.tsx [hunk 1]
```diff
@@ -265,6 +265,7 @@ class InsightViewCore extends React.Component<IInsightViewProps & WrappedCompone
                         backend={this.props.backend}
                         colorPalette={this.props.colorPalette ?? this.state.colorPalette}
                         config={this.props.config}
+                        execConfig={this.props.execConfig}
```
Comment (no23reason) [inline libs/sdk-ui-ext/src/insightView/InsightView.tsx:268]: What about DashboardVIew, should we support it there too?

File: libs/sdk-ui-geo/src/GeoPushpinChart.tsx [hunk 2]
```diff
@@ -92,7 +92,7 @@ const NON_CORE_PROPS: Array<keyof IGeoPushpinChartProps> = [
 ];
 
 function GeoPushpinChartInner(props: IGeoPushpinChartProps): JSX.Element {
-    const { backend, workspace, sortBy, filters, exportTitle } = props;
+    const { backend, workspace, sortBy, filters, exportTitle, execConfig = {} } = props;
```
Comment (no23reason) [inline libs/sdk-ui-geo/src/GeoPushpinChart.tsx:95]: The dafault to {} can cause problems, please remove it  (similarly to the other charts).

File: libs/sdk-ui-pivot/src/PivotTable.tsx [hunk 3]
```diff
@@ -46,14 +46,15 @@ import { AVAILABLE_TOTALS } from "./impl/base/constants";
  * @returns new prepared execution
  */
 function prepareExecution(props: IPivotTableProps): IPreparedExecution {
-    const { backend, workspace, filters, sortBy = [] } = props;
+    const { backend, workspace, filters, sortBy = [], execConfig = {} } = props;
```
Comment (no23reason) [inline libs/sdk-ui-pivot/src/PivotTable.tsx:49]: The dafault to {} can cause problems, please remove it (similarly to the other charts).

File: libs/sdk-ui-geo/src/GeoPushpinChart.tsx (hunk 2 above)
Comment (Tulean) [reply libs/sdk-ui-geo/src/GeoPushpinChart.tsx:95]: unfortunately linting complains about it being undefined.

File: libs/sdk-ui-pivot/src/PivotTable.tsx (hunk 3 above)
Comment (Tulean) [reply libs/sdk-ui-pivot/src/PivotTable.tsx:49]: unfortunately linting complains about it being undefined if you want it to be removed.

File: libs/sdk-ui-ext/src/insightView/InsightView.tsx (hunk 1 above)
Comment (Tulean) [reply libs/sdk-ui-ext/src/insightView/InsightView.tsx:268]: IMO we don't need to support it in dashboard view as in the current sampling does not get saved, therefore it is not propagated in KD.

Comment (Tulean) [reply libs/sdk-ui-ext/src/insightView/InsightView.tsx:268]: let me confirm it with Ondra Macek

File: libs/sdk-ui-geo/src/GeoPushpinChart.tsx (hunk 2 above)
Comment (no23reason) [reply libs/sdk-ui-geo/src/GeoPushpinChart.tsx:95]: Oh, right, there is different strict setting. Ok then

File: libs/sdk-ui-ext/src/insightView/InsightView.tsx (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-ext/src/insightView/InsightView.tsx:268]: But you might want to see a dashboard with a sampling enabled on all insight for example

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4495/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6115/ : FAILURE in 4m 00s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4524/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6137/ : SUCCESS in 14m 14s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1551

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1286/ : SUCCESS in 1m 08s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1562/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/335/ : SUCCESS in 12m 05s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6145/ : SUCCESS in 14m 07s

**Pull request merged**